_test/bootstrap.sh

```diff
@@ -1,27 +1,24 @@
 #!/bin/bash
 # This installs a the prerequisites for a given GHC/Cabal config
-set -x
 set -e
 if [ ! -z "$GHCVER" ]; then
     export PATH=/opt/ghc/${GHCVER}/bin:$PATH
 fi
 if [ ! -z "$CABALVER" ]; then
     export PATH=/opt/cabal/${CABALVER}/bin:$PATH
 fi
-cabal update
-# This is a fairly minimal set
-cabal install \
-      primitive \
-      random \
-      tf-random \
-      HUnit \
-      QuickCheck \
-      split \
-      text \
-      bytestring \
-      attoparsec \
-      vector \
-      parallel \
-      stm \
-      old-locale \
-      lens
+
+# What directory is this script in?
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+deps=$($DIR/dependencies $(ls $DIR/../exercises))
+already_installed=$(ghc-pkg list --names-only --simple-output | tr " " "\n" | sort)
+not_installed=$(comm -23 <(echo "$deps") <(echo "$already_installed"))
+
+echo "Not installed packages ($(echo "$not_installed" | wc -l)):"
```

Comment:
sigh, off by one error. `Not installed packages (1):` when `not_installed` == "'`